src/Reaction.C

```diff
@@ -680,9 +680,15 @@ Reaction::operator!=(const Reaction & other) const
 const string
 Reaction::getReferencesAsString() const
 {
-  string temp_refs = "";
-  for (auto r : _references)
-    temp_refs += "\\cite{" + r + "}";
+  string temp_refs = "\\cite{";
+  for (size_t r = 0; r < _references.size(); r++) {
+    temp_refs += _references[r];
+	if (r != _references.size() - 1)
+		temp_refs += ",";
+  }
+  temp_refs += "}";
+  if (temp_refs == "\\cite{}")
+	  temp_refs = "";
```

Comment:
```suggestion
  string temp_refs = "\\cite{";
  for (size_t i = 0; i < _references.size(); i++) {
    temp_refs += _references[i];
	if (i != _references.size() - 1)
		temp_refs += ", ";
  }
  temp_refs += "}";
```

Since we are not iterating over the objects in the _references vector directly now the loop index should be `i` just for the sake of consistency. Additionally, we will never have a Reaction that has no references since we force all Reactions to have a reference so there is no need for the check at the bottom of the loop.